app/controllers/candidate_interface/degrees/destroy_controller.rb

```diff
@@ -2,22 +2,22 @@ module CandidateInterface
   class Degrees::DestroyController < CandidateInterfaceController
     def confirm_destroy
       application_form = current_candidate.current_application
-      @degree = DegreesForm.find_by_application(application_form, degrees_params[:id])
+      @degree = DegreeForm.build_from_application(application_form, degree_id_params)
     end
 
     def destroy
       current_candidate.current_application
         .application_qualifications
-        .find(degrees_params[:id])
+        .find(degree_id_params)
         .destroy!
 
       redirect_to candidate_interface_degrees_review_path
     end
 
   private
 
-    def degrees_params
-      params.permit(:id)
+    def degree_id_params
```

Comment:
`params` is misleading if it returns a single value — maybe `current_degree_id`?